File: Dot examples/ASP Book Examples/Book Examples/Chapter31/vb/App_Code/Movie.vb
Imports System
Imports System.Collections
Imports System.Data
Imports System.Data.SqlClient
Imports System.Web.Configuration

Public Class Movie
    Public Shared Function Search(ByVal query As String) As ArrayList
        Dim results As New ArrayList()
        Dim conString As String = WebConfigurationManager.ConnectionStrings("con").ConnectionString
        Dim commandText As String = "SELECT Title FROM Movie WHERE Title LIKE @query + '%'"
        Dim con As New SqlConnection(conString)
        Dim cmd As New SqlCommand(commandText, con)
        cmd.Parameters.AddWithValue("@query", query)
        Using con
            con.Open()
            Dim reader As SqlDataReader = cmd.ExecuteReader()
            While reader.Read()
                results.Add(CType(reader("Title"), String))
            End While
        End Using
        Return results
    End Function


End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter31/cs/App_Code/Movie.cs
using System;
using System.Collections;
using System.Data;
using System.Data.SqlClient;
using System.Web.Configuration;

public class Movie
{
    public static ArrayList Search(string query)
    {
        ArrayList results = new ArrayList();
        string conString = WebConfigurationManager.ConnectionStrings["con"].ConnectionString;
        string commandText = "SELECT Title FROM Movie WHERE Title LIKE @query + '%'";
        SqlConnection con = new SqlConnection(conString);
        SqlCommand cmd = new SqlCommand(commandText, con);
        cmd.Parameters.AddWithValue("@query", query);
        using (con)
        {
            con.Open();
            SqlDataReader reader = cmd.ExecuteReader();
            while (reader.Read())
                results.Add((string)reader["Title"]);
        }
        return results;
    }


}
